archaius2-core/src/main/java/com/netflix/archaius/config/PollingDynamicConfig.java

```diff
@@ -156,11 +163,6 @@ public void forEachPropertyUninstrumented(BiConsumer<String, Object> consumer) {
         current.forEach(consumer);
     }
 
-    public void setAccessMonitorUtil(AccessMonitorUtil accessMonitorUtil) {
-        this.accessMonitorUtil = accessMonitorUtil;
-        instrumentationEnabled = true;
-    }
-
     @Override
     public void recordUsage(PropertyDetails propertyDetails) {
         if (instrumentationEnabled()) {
```

Comment:
Let's just put a note here for when we come up with the runtime-disabling mechanism. If it includes a way to change the AccessMonitorUtil object, then we'll still have a window for a race (if it is not null at the point of this check and it gets unset before we get to the `registerUsage` call. In that case we should probably move the body of this method to live inside the `Instrumentation` object, which would get around the possible issue.
But for now it's probably best to leave it like this, it's probably easier to read.